src/Contracts/MultiSearchFederation.php

```diff
@@ -16,6 +16,13 @@ class MultiSearchFederation
      */
     private ?int $offset = null;
 
+    /**
+     * @var array<non-empty-string, list<non-empty-string>>|null
+     */
+    private ?array $facetsByIndex = null;
+
+    private ?array $mergeFacets = null;
```

Comment:
What's the array shape?

Reply:
string -> any value

Reply:
list or hash? then it should be documented as `@var list<string>` or `@var array<string>` :)

Reply:
```suggestion
    /**
     * @var array<non-empty-string>|null
     */
    private ?array $mergeFacets = null;
```

something like this?

Reply:
yes :) i found this:

![image](https://github.com/user-attachments/assets/29334a8a-dc9f-456a-9492-e8538a97d5f1)

so it can be also an integer, right?

Reply:
mergeFacets is an object with only one field so far, but we should be able to support any kind of value depending on the evolution of this setting

Reply:
just thinking out loud: should `mergeFacets` be documented as `array{maxValuesPerFacet: int}` as looking at the implementation PR there is only this single option

Reply:
yes, gh didn't show your comment before posting mine :) we can evolve php docs after new features are merged.
with this shape definition user will get IDE autocompletion :)

Reply:
and if the settings ios changed, we would just change this into `array{maxValuesPerFacet: int, newSetting: string}`?

Reply:
we could even make a new DTO for `mergeFacets`

Reply:
> and if the settings ios changed, we would just change this into:
> 
> ```
> array{maxValuesPerFacet: int, newSetting: string}
> ```
> 
> ?

yes

Reply:
but I'd prefer a new DTO :)

Reply:
Ok so to summarize, for now we add the following code:
```suggestion
    /**
     * @var array{maxValuesPerFacet: int}|null
     */
    private ?array $mergeFacets = null;
```

and if, in the future, the `mergeFacets` becomes more complex, we will create a DTO.

Reply:
Ok

Reply:
just a nitpick, but maybe `@var array{maxValuesPerFacet: positive-int}|null`